hwy/tests/widen_mul_test.cc

```diff
@@ -493,6 +493,105 @@ HWY_NOINLINE void TestAllReorderWidenMulAccumulate() {
   ForShrinkableVectors<TestReorderWidenMulAccumulate>()(uint16_t());
 }
 
+struct TestWidenMulAccumulate {
+  template <typename TN, class DN>
+  HWY_INLINE void operator()(TN /*unused*/, DN dn) {
+    using TW = MakeWide<TN>;
+    const RepartitionToWide<DN> dw;
+    const Half<DN> dnh;
+    using VW = Vec<decltype(dw)>;
+    using VN = Vec<decltype(dn)>;
+    const size_t NN = Lanes(dn);
+
+    VW f0 = Zero(dw);
+    const VN bf0 = Zero(dn);
+    const VN bf1 = Set(dn, TW{5});
+
+    // Any input zero => both outputs zero
+    VW sum1 = f0;
+    HWY_ASSERT_VEC_EQ(dw, f0,
+                      WidenMulAccumulate(dn, bf0, bf0, f0, sum1));
+    HWY_ASSERT_VEC_EQ(dw, f0, sum1);
+    HWY_ASSERT_VEC_EQ(dw, f0,
+                      WidenMulAccumulate(dn, bf0, bf1, f0, sum1));
+    HWY_ASSERT_VEC_EQ(dw, f0, sum1);
+    HWY_ASSERT_VEC_EQ(dw, f0, WidenMulAccumulate(dn, bf1, bf0, f0, sum1));
+    HWY_ASSERT_VEC_EQ(dw, f0, sum1);
+
+    // delta[p] := 1, all others zero.
+    auto delta_w = AllocateAligned<TW>(NN);
+    for (size_t p = 0; p < NN; ++p) {
+      // Workaround for incorrect Clang wasm codegen: re-initialize the entire
+      // array rather than zero-initialize once and then toggle lane p.
+      for (size_t i = 0; i < NN; ++i) {
+        delta_w[i] = static_cast<TW>(i == p);
+      }
+      const VW delta0 = Load(dw, delta_w.get());
+      const VW delta1 = Load(dw, delta_w.get() + NN / 2);
+      const VN delta = Combine(dn, DemoteTo(dnh, Add(delta0, delta1)), DemoteTo(dnh, Sub(delta0, delta1)));
```

Comment:
I think we could just use OrderedDemote2To?

Reply:
OrderedDemote2To is not supported for f16. So that the reason I changed to this one. I can create another test for f16 if required

Reply:
Ah, makes sense. Let's leave as-is :)